apps/drec-api/src/pods/device-group/device-group.entity.ts

```diff
@@ -16,5 +30,72 @@ export class DeviceGroup extends ExtendedBaseEntity implements IDeviceGroup {
   @Column()
   organizationId: number;
 
+  @Column()
+  @IsString()
+  fuelCode: string;
+
+  @Column()
+  @IsString()
+  countryCode: string;
+
+  @Column()
+  @IsEnum(StandardCompliance)
+  standardCompliance: StandardCompliance;
+
+  @Column('text', { array: true })
+  deviceTypeCodes: string[];
+
+  @Column({
+    type: 'enum',
+    enum: OffTaker,
+    array: true,
+    default: [],
+  })
+  offTakers: OffTaker[];
+
+  @Column({
+    type: 'enum',
+    enum: Installation,
+    array: true,
+    default: [],
+  })
+  installationConfigurations: Installation[];
+
+  @Column({
+    type: 'enum',
+    enum: Sector,
+    array: true,
+    default: [],
+  })
+  sectors: Sector[];
+
+  @Column({
+    type: 'enum',
+    enum: CommissioningDateRange,
+    array: true,
+    default: [],
+  })
+  commissioningDateRange: CommissioningDateRange[];
+
+  @Column()
+  @IsBoolean()
+  gridInterconnection: boolean;
+
+  @Column()
+  @IsNumber()
+  aggregatedCapacity: number;
+
+  @Column()
+  @IsEnum(CapacityRange)
+  capacityRange: CapacityRange;
+
+  @Column({ default: 1000 })
+  @IsNumber()
+  yieldValue: number;
```

Comment:
And this is not compatible with DeviceGroup type which has yieldValue optional.

Reply:
Well, when creating a group, the yield value is not required and if it is missing, then we use the default which is 1000.

Reply:
So it will always be defined, but the DeviceGroup interface  tells, that yieldValue is optional. It will never be optional, it will always be defined.

Or I misinterpet DeviceGroup usage
Or DeviceGroup interface is used for too many things

Reply:
The interface is used as inheritance for Dtos and entity. Dtos have it optional but in the entity, if it`s not there, we make sure it is by setting it. I understand it seems a bit weird, but we will get most of this data from crawiling multiple API`s and I doubt all devices will have yield value. I can`t have it in the entity optional but I can`t enforce it in either in the Dtos.

Reply:
I'm super strict about my types, that's probably why I care about this :p If you are ok with this so am I